packages/cli/src/commands/deploy-v2.ts

```diff
@@ -0,0 +1,111 @@
+import chalk from "chalk";
+import { ethers } from "ethers";
+import { execa } from "execa";
+import glob from "glob";
+import { basename } from "path";
+import type { CommandModule } from "yargs";
+import { loadWorldConfig, ResolvedWorldConfig } from "../config/loadWorldConfig.js";
+import { deploy } from "../utils/deploy-v2.js";
+import { getOutDirectory } from "../utils/forgeConfig.js";
+
+type Options = {
+  configPath?: string;
+  print?: boolean;
+};
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "deploy-v2",
+
+  describe: "Deploy MUD v2 contracts",
+
+  builder(yargs) {
+    return yargs.options({
+      configPath: { type: "string", desc: "Path to the config file" },
+      print: { type: "boolean", desc: "Print the resolved config" },
+    });
+  },
+
+  async handler({ configPath, print }) {
+    const config = await loadWorldConfig(configPath);
+
+    // Run forge build
+    try {
+      await execa("forge", ["build"], { stdout: "inherit", stderr: "pipe" });
+    } catch (error: any) {
+      if (error?.stderr) {
+        console.error(error.stderr);
+      }
+      console.error(chalk.red("Error running `forge build`"));
+      process.exit(1);
+    }
+
+    // Get a list of all contracts in output ending with `System`
+    const outDir = await getOutDirectory();
+    const defaultSystemNames = glob
+      // Get all files in outDir ending in `*System.sol  `
+      .sync(`${outDir}/*System.sol`)
```

Comment:
should this be `${outDir}/**/*System.sol` to also capture system files in subdirectories?

Reply:
It seems like forge flattens the files in the output directory - in my test project my system is at `src/systems/IncrementSystem.sol`, the output is at `out/IncrementSystem.sol`

Reply:
oh look, I've already reported this 🙈  https://github.com/foundry-rs/foundry/issues/2079